library/ccm.c

```diff
@@ -158,7 +158,13 @@ static int ccm_auth_crypt( mbedtls_ccm_context *ctx, int mode, size_t length,
      * 'length' checked later (when writing it to the first block)
      */
     if( tag_len < 4 || tag_len > 16 || tag_len % 2 != 0 )
```

Comment:
Please simplify this test not to have nested if's. Eg..

    if( tag_len == 2 || tag_len > 16 || tag_len % 2 != 0 )